File: Easy/Last Stone Weight.py
class Solution:
    def lastStoneWeight(self, stones: List[int]) -> int:
        heapq.heapify(stones)
        k = 1
        while len(stones) > 1:
            twoHeaviest = heapq.nlargest(2, stones)

            if twoHeaviest[0] != twoHeaviest[1]:
                stones.append(abs(twoHeaviest[0] - twoHeaviest[1]))
                
            stones.remove(twoHeaviest[0])
            stones.remove(twoHeaviest[1])
            heapq.heapify(stones)
        
        if stones:
            return stones[-1]
        return 0
            

File: LeetCode/C#/Easy/Last Stone Weight.cs
public class Solution {
    public int LastStoneWeight(int[] stones) {
        var sl = stones.ToList();
        
        int first = 0, second = 0, smash = 0;
        
        while(sl.Count > 1) {
            first = sl.Max(); sl.Remove(sl.Max());
            second = sl.Max(); sl.Remove(sl.Max());
            
            smash = Math.Abs(first - second);
            if (smash != 0) 
                sl.Add(smash);
        }
        
        return sl.Count > 0 ? sl[0] : 0;
        
        
    }
}